sql-plugin/src/main/scala/org/apache/spark/sql/rapids/stringFunctions.scala

```diff
@@ -905,7 +912,10 @@ class GpuRLikeMeta(
         case Literal(str: UTF8String, DataTypes.StringType) if str != null =>
           try {
             // verify that we support this regex and can transpile it to cuDF format
-            pattern = Some(new CudfRegexTranspiler(RegexFindMode).transpile(str.toString, None)._1)
+            val transpiledAST = 
+                new CudfRegexTranspiler(RegexFindMode).getTranspiledAST(str.toString, None)._1 
```

Comment:
I think you can decompose this a bit cleaner actually

```suggestion
            val (transpiledAST, _) = 
                new CudfRegexTranspiler(RegexFindMode).getTranspiledAST(str.toString, None)
```